File: serial_lib/src/main/java/com/cl/serialportlibrary/listener/OnSerialPortDataListener.java
package com.cl.serialportlibrary.listener;

import com.cl.serialportlibrary.enumerate.SerialPortEnum;

/**
 * 串口消息监听
 */
public interface OnSerialPortDataListener {

    /**
     * 数据接收
     *
     * @param bytes 接收到的数据
     */
    void onDataReceived(byte[] bytes, SerialPortEnum serialPortEnum);

    /**
     * 数据发送
     *
     * @param bytes 发送的数据
     */
    void onDataSent(byte[] bytes,SerialPortEnum serialPortEnum);

}

File: serial_lib/src/main/java/com/cl/serialportlibrary/listener/OnSerialPortDataListener.kt
package com.cl.serialportlibrary.listener

import com.cl.serialportlibrary.enumerate.SerialPortEnum

interface OnSerialPortDataListener {
    fun onDataReceived(bytes: ByteArray, serialPortEnum: SerialPortEnum)

    fun onDataSent(bytes: ByteArray, serialPortEnum: SerialPortEnum)
}
